var/spack/repos/builtin/packages/glibc/package.py

```diff
@@ -155,22 +199,25 @@ def patch(self):
             string=True,
         )
 
-    depends_on("bison", type="build")
-    depends_on("texinfo", type="build")
-    depends_on("gettext", type="build")
-    depends_on("perl", type="build")
-    depends_on("gawk", type="build")
-    depends_on("sed", type="build")
-    depends_on("gmake", type="build")
+    # This is an absolutely wretched hack to allow building a glibc that doesn't pollute
+    # the new environment.  It should go away as soon as we have a way.
+    with when("~stage1"):
+        depends_on("bison", type="build")
+        depends_on("texinfo", type="build")
+        depends_on("gettext", type="build", when="~stage1")
+        depends_on("perl", type="build")
+        depends_on("gawk", type="build")
+        depends_on("sed", type="build")
+        depends_on("gmake", type="build")
 
-    # See 2d7ed98add14f75041499ac189696c9bd3d757fe
-    depends_on("gmake@:4.3", type="build", when="@:2.36")
+        # See 2d7ed98add14f75041499ac189696c9bd3d757fe
+        depends_on("gmake@:4.3", type="build", when="@:2.36")
 
-    # From 2.29: generates locale/C-translit.h
-    # before that it's a test dependency.
-    depends_on("python@3.4:", type="build", when="@2.29:")
+        # From 2.29: generates locale/C-translit.h
+        # before that it's a test dependency.
+        depends_on("python@3.4:", type="build", when="@2.29:")
 
-    depends_on("linux-headers")
+    depends_on("linux-headers", type="build")
```

Comment:
This was `link` type before s.t. you'd get automatic `-I` flags. Can you leave a comment that this is no longer necessary cause the headers are copied?

Reply:
Hm, that’s a good point. I can definitely add the comment at least. Worth some testing.